notebooks/WFC3/point_spread_function/psf_utilities.py

```diff
@@ -0,0 +1,402 @@
+    matplotlib.pyplot.savefig(filename)
+    if (show_figure is True):
+        matplotlib.pyplot.show()
+    if (show_figure is False):
+        matplotlib.pyplot.close()
+
+    return print('\nFigure saved as: '+filename)
+
+
+def setup_matplotlib(size, multiplier):
+
+    '''
+    Change the default matplotlib parameters to improve figures.
+    The size parameter sets the overall image size, while the
+    multiplier parameter increases the fontsize and line widths.
+    '''
+
+    matplotlib.pyplot.rcdefaults() # restore default values
+    matplotlib.pyplot.rcParams["font.family"] = "STIXGeneral"
+    matplotlib.pyplot.rcParams["mathtext.fontset"] = "stix"
+    matplotlib.pyplot.rcParams['text.usetex'] = False
+    default_thickness_frame = 0.66
+    default_thickness_lines = 1.00
+    default_fontsize = 10
+    
+    if (size == 'notebook'):
+        my_scale = 1.0
+        my_figure_size = (11.0, 11.0)
+    else:
+        my_scale = multiplier
+        my_figure_size = (size, size)
+        
+    my_scale = (my_scale * multiplier)
+    matplotlib.pyplot.rcParams["figure.figsize"] = my_figure_size
+    matplotlib.pyplot.rcParams["font.size"] = default_fontsize * my_scale
+    matplotlib.pyplot.rcParams["axes.linewidth"] = default_thickness_frame * my_scale
+    matplotlib.pyplot.rcParams["lines.linewidth"] = default_thickness_lines * my_scale
+    matplotlib.pyplot.rcParams['xtick.major.width'] = default_thickness_frame * my_scale
+    matplotlib.pyplot.rcParams['xtick.minor.width'] = default_thickness_frame * my_scale
+    matplotlib.pyplot.rcParams['ytick.major.width'] = default_thickness_frame * my_scale
+    matplotlib.pyplot.rcParams['ytick.minor.width'] = default_thickness_frame * my_scale
+    matplotlib.pyplot.rcParams['xtick.major.size'] = 3.0 * my_scale
+    matplotlib.pyplot.rcParams['ytick.major.size'] = 3.0 * my_scale
+    matplotlib.pyplot.rcParams['xtick.minor.size'] = 1.5 * my_scale
+    matplotlib.pyplot.rcParams['ytick.minor.size'] = 1.5 * my_scale
+    my_fontsize = matplotlib.pyplot.rcParams["font.size"]
+        
+    return my_figure_size, my_fontsize
+
+
+def create_mask(data, cutout_size, xcenter, ycenter):
+
+    '''
+    A function that accepts an array of image data, a cutout size 
+    in pixels, and xcenter and ycenter locations. The function returns 
+    a mask that is compatible with the daofind function so that source 
+    searches are only conducted on a portion of the overall image.
+    '''
+    
+    # Create a mask array that is the same size as the science image and mask the upper, lower, left, and right regions.
+    mask = numpy.zeros(data.shape, dtype=bool)
+    mask[ycenter+int(cutout_size/2):numpy.shape(data)[0], 0:numpy.shape(data)[1]] = True
+    mask[0:ycenter-int(cutout_size/2), 0:numpy.shape(data)[1]] = True
+    mask[0:numpy.shape(data)[0], 0:xcenter-int(cutout_size/2)] = True
+    mask[0:numpy.shape(data)[0], xcenter+int(cutout_size/2):numpy.shape(data)[1]] = True
+
```

Comment:
I think this may have the opposite effect. Specifically, each of the four lines is setting the top, bottom, left, and right regions _outside_ of the box to be True, while the refactored version is setting the values _inside_ the box to True?

Alternatively, I could set the entire array to True, and then use the line you provided to set the inside to False?

Reply:
Oh I see, then the alternative version would be:
```
mask = np.ones(data.shape, dtype=bool)
size = int(cutout_size / 2)
mask[ycenter-size:ycenter+size, xcenter-size:xcenter+size] = False
```
Up to you if you want to update the coding block 😃

Reply:
Done.